app/controllers/items_controller.rb

```diff
@@ -0,0 +1,42 @@
+class ItemsController < ApplicationController
+
+  def index
+    @items = Item.all
+  end
+
+  def show
+    @item = Item.find(params[:id])
+  end
+
+  def new
+    @merchant_id = params[:merchant_id]
+  end
+
+  def edit
+    @item_id = params[:id]
+  end
+
+  def update
+    Item.find(params[:id]).update(item_params)
+    redirect_to "/items/#{params[:id]}"
+  end
```

Comment:
Since your form does not prepopulate any data from that item you're trying to update, it will only update if you pass it all of the params. Prepopulating can be done in your forms, and I think you had asked a question about that during our meeting the other day so you probably know that now, but I'm writing this anyway haha